tcms/testcases/tests/test_views.py

```diff
@@ -228,6 +232,82 @@ def _assertTestCase(self, test_case):
         self.assertEqual(test_case.notes, self.notes)
 
 
+class TestNewCasePermission(PermissionsTestCase):
+    permission_label = 'testcases.add_testcase'
+    http_method_names = ['get', 'post']
+    url = reverse('testcases-new')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.summary = 'summary'
+        cls.post_data = {
+            'summary': cls.summary,
+        }
+        super().setUpTestData()
+        user_should_have_perm(cls.tester, 'testcases.view_testcase')
+
+        cls.case_status_confirmed = TestCaseStatus.objects.get(name='CONFIRMED')
+
+        cls.case = TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=cls.case_status_confirmed,
+        )
+        cls.case.save()
+
+        cls.post_data.update({
+            'author': cls.tester.pk,
+            'default_tester': cls.tester.pk,
+            'priority': cls.case.priority.pk,
+            'product': cls.case.category.product.pk,
+            'category': cls.case.category.pk,
+            'case_status': cls.case_status_confirmed.pk,
+            'text': 'some text description',
+            'script': 'some script',
+            'arguments': 'args1, args2, args3',
+            'email_settings-0-auto_to_case_author': 'on',
+            'email_settings-0-auto_to_run_manager': 'on',
+            'email_settings-0-auto_to_case_run_assignee': 'on',
+            'email_settings-0-auto_to_case_tester': 'on',
+            'email_settings-0-auto_to_run_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_case_delete ': 'on',
+            'email_settings-0-cc_list': 'info@example.com',
+            'email_settings-0-case': '',
+            'email_settings-0-id': cls.case.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        })
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _('New TestCase'))
+
+    def verify_post_with_permission(self):
+        response = self.client.post(self.url, self.post_data, follow=True)
+        test_case = TestCase.objects.get(summary=self.summary)
+        redirect_url = reverse('testcases-get', args=[test_case.pk])
+
+        self.assertRedirects(response, redirect_url, status_code=302, target_status_code=200)
+        self._assertTestCase(test_case)
```

Comment:
This method is overkill here. Line 293 above makes sure the newly created record is in the database and then the assertRedirects verifies that the browser was redirected correctly. At most you can check that the summary is displayed on the new page (isn't strictly necessary). The rest basically validates the same.

Reply:
Removed the unnecessary part.